src/hotspot/share/gc/g1/g1BlockOffsetTable.hpp

```diff
@@ -72,7 +72,9 @@ class G1BlockOffsetTable: public CHeapObj<mtGC> {
 
   inline void set_offset_array(size_t left, size_t right, u_char offset);
 
+#ifdef ASSERT
```

Comment:
I disagree with this change - it is common usage to use the `NOT_DEBUG_RETURN` or similar macros to hide the `#ifdef`s.
The `#ifdef`s only clutter the code for no gain as you can see.

Reply:
I agree with @tschatzl - don't make this and related parts of the change.

Reply:
Got it. Thanks for your guide.